pkg/config/nodetreemodel/config.go

```diff
@@ -190,6 +206,48 @@ func (c *ntmConfig) GetKnownKeysLowercased() map[string]interface{} {
 	return ret
 }
 
+// SetReady is called when Setup is complete, and the config is ready to be used
+func (c *ntmConfig) SetReady() {
+	defaultSourceRoot, err := NewNode(c.defaultBuilder, model.SourceDefault)
+	if err != nil {
+		log.Errorf("%s", err)
+	}
+	// Assign the built tree to the default source
+	c.defaultSourceRoot = defaultSourceRoot
+	// By clearing the builder, the config is considered "Ready" for use
+	// Furthur calls to SetKnown, BindEnv, SetDefault will panic
+	c.defaultBuilder = nil
+	// TODO: Build the environment variable tree
+	// TODO: Instead of assigning defaultSource to root, merge the trees
+	c.root = c.defaultSourceRoot
+}
+
+func (c *ntmConfig) isReady() bool {
+	return c.defaultBuilder == nil
+}
+
+func (c *ntmConfig) insertDefaultValue(key string, defaultValue interface{}) {
+	currNode := c.defaultBuilder
+	keyParts := strings.Split(key, ".")
+	for len(keyParts) > 0 {
+		first := keyParts[0]
+		keyParts = keyParts[1:]
+		if currNode[first] == nil && len(keyParts) == 0 {
+			// Assign the leaf value if it used to be nil (don't overwrite existing defaults)
+			currNode[first] = defaultValue
+			break
+		} else if currNode[first] == nil {
+			build := make(map[string]interface{})
+			currNode[first] = build
+			currNode = build
+		} else if conv, ok := currNode[first].(map[string]interface{}); ok {
+			currNode = conv
+		} else {
+			return
+		}
```

Comment:
This should only happened when setting conflicting default, right ? We might want to handle the error.